src/ray/common/test_util.cc

```diff
@@ -23,11 +23,27 @@
 namespace ray {
 
 void RedisServiceManagerForTest::SetUpTestCase() {
-  auto seed = std::chrono::high_resolution_clock::now().time_since_epoch().count();
-  std::mt19937 gen(seed);
-  std::uniform_int_distribution<int> random_gen{2000, 7000};
-  // Use random port to avoid port conflicts between UTs.
-  REDIS_SERVER_PORT = random_gen(gen);
+  std::vector<int> ACTUAL_REDIS_SERVER_PORTS;
+  if (REDIS_SERVER_PORTS.empty()) {
+    ACTUAL_REDIS_SERVER_PORTS.push_back(StartUpRedisServer(0));
+  } else {
+    for (const auto &port : REDIS_SERVER_PORTS) {
+      ACTUAL_REDIS_SERVER_PORTS.push_back(StartUpRedisServer(port));
+    }
+  }
+  REDIS_SERVER_PORTS = ACTUAL_REDIS_SERVER_PORTS;
+}
+
+// start a redis server with specified port, use random one when 0 given
+int RedisServiceManagerForTest::StartUpRedisServer(int port) {
+  int actual_port = port;
+  if (port == 0) {
+    auto seed = std::chrono::high_resolution_clock::now().time_since_epoch().count();
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<int> random_gen{2000, 7000};
```

Comment:
nit: it's not a big deal, but I think just use rand() should suffice for the randomization here?  `std::mt19937` is only required when we need to general a large number of unique values

Reply:
the original codes use std::mt19937 so i just follow them, but i agree with you. let's try to simplify it. :)